### ISSUE TITLE
Questions Regarding Evaluation

### ISSUE THREAD

**Issue by gmltmd789** [ZihanZhaoSJTU/LibriSQA#1]
Hello,

Thank you for providing high-quality data.

I have some questions regarding the code for measuring metrics (BERT-Score, F1 Score, Accuracy, Rouge (1, 2, L), BLEU-score) between the generated text and the ground truth answer.

I am curious about which libraries were used to measure each metric.

In the case of the BERT score, I am wondering whether RoBERTa, BERT, or another model was used.

I am curious about how the generated text and the original text were processed before measuring each metric. The current text is case-sensitive and contains punctuation, and I am wondering how this was handled.

Thank you once again for the high-quality data.

**Comment by ZihanZhaoSJTU** [comment]
Hello,

Thank you very much for your interest in our project and for your appreciation of the data we provide. We are happy to answer your queries.

Regarding your first question about the metrics, we utilized the torchmetrics library to calculate the F1 Score, Accuracy, Rouge (1, 2, L), and BLEU-score. As for the BERT-Score, it was computed using our own written code.

In response to your query about which model was used for the BERT score, we employed the BERT model.

Regarding your question about how we processed the generated text and the original text before measuring each metric, we implemented some preprocessing steps. Specifically, we removed all punctuation and converted all characters to lowercase. This was done to reduce unnecessary variability and to focus on the core meaning of the text content.

Should you have any further questions or require more detailed information, please do not hesitate to contact us.

**Comment by gmltmd789** [comment]
Thank you for your response. Is the BLEU Score the standard BLEU, or is it SacreBLEU? 
For BERT score, did you use "bert-base-uncased" in huggingface?
And did you remove '? e.g., can't -> cant

**Comment by ZihanZhaoSJTU** [comment]
We did use the standard BLEU, "bert-base-uncased" in huggingface. We removed all the punctuation for simplicity, and since the situation you mentioned is not particularly common in the dataset, we think it won't have a big impact on the results.

**Comment by gmltmd789** [comment]
Thank you for your kind response!

**Comment by gmltmd789** [comment]
Hello,

I have some additional questions regarding the model evaluation.

To ensure a comparable evaluation in the same context, I am attaching my evaluation code.

1. Initially, I implemented the evaluation code using the following approach, and I would appreciate any advice or comments you may have on this matter.

2. In the paper, it mentions BLEU-1. I am curious whether this refers to 1-gram BLEU or if it considers BLEU for 1 to 4 grams collectively. Could you please clarify which values it specifically denotes?

3. Regarding the BERT similarity metric, I implemented the code to measure similarity using BERT-based uncased, similar to the SentenceBERT paper you referred to. Is it appropriate to measure it as outlined below, or did you follow additional steps such as fine-tuning to measure it with a custom model, as described in SentenceBERT?

Thank you for your kind response!

    import json
    import numpy as np
    import evaluate
    import string
    import torch
    from sentence_transformers import SentenceTransformer, util
    from tqdm import tqdm
    
    predictions = # load your predictions
    references = # load your references
    
    # preprocess_text: remove all punctuation and lower the sentence.
    predictions = [preprocess_text(i) for i in predictions]
    references = [preprocess_text(i) for i in references]

    bleu = evaluate.load("bleu")

    results = bleu.compute(predictions=predictions, references=[[i] for i in references])

    print(f"BLEU Score: {results['bleu']}")

    rouge = evaluate.load('rouge')
    results = rouge.compute(predictions=predictions, references=references)

    print(f"Rouge Score-1: {results['rouge1']}")
    print(f"Rouge Score-2: {results['rouge2']}")
    print(f"Rouge Score-L: {results['rougeL']}")

    # Load BERT model w/ sentence BERT
    model = SentenceTransformer('bert-base-uncased')

    bert_scores = []
    for prediction, reference in tqdm(zip(predictions, references)):
        with torch.no_grad():
            embedding1 = model.encode(prediction, convert_to_tensor=True)
            embedding2 = model.encode(reference, convert_to_tensor=True)
        # cosine similarity calculation
        cosine_similarity = util.pytorch_cos_sim(embedding1, embedding2)
        bert_scores.append(cosine_similarity.cpu().item())

    print(f"BERT Score: {np.array(bert_scores).mean()}")`

**Comment by gmltmd789** [comment]
I inquired again via email, and they confirmed that BLEU is calculated using 1-gram BLEU, and BERT similarity is accurate! Thank you for your kind response.
